uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/components/manage-leave-requests.spec.js

```diff
@@ -1,929 +1,953 @@
-              var extractStartDate = function (period) {
-                return period.start_date;
-              };
-              var absencePeriodSortedByDate = _.sortBy(absencePeriodData.all().values, 'start_date').map(extractStartDate);
+        it('locations have loaded', function () {
+          expect(controller.locations).toEqual(optionGroupMock.getCollection('hrjc_location'));
+        });
 
-              expect(controller.absencePeriods.map(extractStartDate)).toEqual(absencePeriodSortedByDate);
-            });
-          });
+        it('level types have loaded', function () {
+          expect(controller.levelTypes).toEqual(optionGroupMock.getCollection('hrjc_level_type'));
+        });
 
-          it('absence types have loaded', function () {
-            expect(controller.absenceTypes.length).not.toBe(0);
+        describe('absence periods', function () {
+          it('loads the absence periods', function () {
+            expect(controller.absencePeriods.length).toBe(absencePeriodData.all().values.length);
           });
 
-          it('filtered list of contacts have loaded', function () {
-            expect(controller.filteredUsers).toEqual(ContactAPIMock.mockedContacts().list);
-          });
+          it('sorts absence periods by start_date', function () {
+            var extractStartDate = function (period) {
+              return period.start_date;
+            };
+            var absencePeriodSortedByDate = _.sortBy(absencePeriodData.all().values, 'start_date').map(extractStartDate);
 
-          it('leave requests have loaded', function () {
-            expect(controller.leaveRequests.table.list.length).toEqual(leaveRequestData.all().values.length);
-            expect(controller.leaveRequests.filter.list.length).toEqual(leaveRequestData.all().values.length);
+            expect(controller.absencePeriods.map(extractStartDate)).toEqual(absencePeriodSortedByDate);
           });
         });
-      });
 
-      describe('pagination', function () {
-        describe('totalNoOfPages', function () {
-          var returnValue;
+        it('absence types have loaded', function () {
+          expect(controller.absenceTypes.length).not.toBe(0);
+        });
 
-          beforeEach(function () {
-            controller.leaveRequests.table.total = 20;
-            controller.pagination.size = 5;
-            returnValue = controller.totalNoOfPages();
-          });
+        it('filtered list of contacts have loaded', function () {
+          expect(controller.filteredUsers).toEqual(ContactAPIMock.mockedContacts().list);
+        });
 
-          it('returns correct number of pages', function () {
-            expect(returnValue).toBe(4);
-          });
+        it('leave requests have loaded', function () {
+          expect(controller.leaveRequests.table.list.length).toEqual(leaveRequestData.all().values.length);
+          expect(controller.leaveRequests.filter.list.length).toEqual(leaveRequestData.all().values.length);
```

Comment:
It's better to have each expect in a different `it` statement in case one fails it's makes finding the error easier.

This one could be something like `it('updates filters for each type of leave request loaded', function() {` or at least that's what I think it's doing.

This also helps explain what's supposed to happen (if anyone reads the specs for that).